Fix image svg

mendyEdri opened the pull request:
## Description
Until now, all images with url used `SvgImage` component, because source was a url and `isSvg` returned `true`.

## Changelog
Added support for `Image` web

## Additional info

OrenZak commented on package.json:
I think its good to add a demo item to the webDemo app

OrenZak replied:
also can you create a version, so we can test it before the merge ? in our Editor
@AmitShwarts, can you give it a try once, @mendyEdri  shares a version ?

mendyEdri replied:
7.8.0-snapshot.3463

AmitShwarts replied:
@mendyEdri as we just talked, we're getting lot's of issues of width, height and resizeMode.
all of the page layout breaks for some reason because of this.. not sure why

OrenZak commented on src/components/image/index.tsx:
```diff
@@ -226,23 +226,31 @@ class Image extends PureComponent<Props, State> {
     const shouldFlipRTL = supportRTL && Constants.isRTL;
     const ImageView = this.shouldUseImageBackground() ? ImageBackground : RNImage;
     const {margins} = modifiers;
+    const imageViewStyle = [
+      tintColor && {tintColor},
+      shouldFlipRTL && styles.rtlFlipped,
+      width && {width},
+      height && {height},
+      cover && styles.coverImage,
+      this.isGif() && styles.gifImage,
+      aspectRatio && {aspectRatio},
+      !useImageInsideContainer && margins,
+      useImageInsideContainer && styles.containImage,
+      style,
+      useImageInsideContainer && styles.shrink
+    ];
```
this wasn't change at all right ?

mendyEdri replied:
Nope

OrenZak commented on src/components/image/index.tsx:
```diff
@@ -226,23 +226,31 @@ class Image extends PureComponent<Props, State> {
+      useImageInsideContainer && styles.shrink
+    ];
+
+    if (!this.shouldUseImageBackground() && Constants.isWeb) {
```
`Constants.isWeb` this will be always false, as you have `index.web.tsx`

OrenZak commented on src/components/image/index.web.tsx:
```diff
@@ -0,0 +1,236 @@
+        <img 
+          {...others} 
+          src={finalSource} 
+          style={StyleSheet.flatten([imageViewStyle, styles.containImage]) as CSSProperties}
```
why do we need to always flat this ?

mendyEdri replied:
Same as you guys did, thought it's for flattening any sort of imageProps passed etc

OrenZak replied:
i'm not sure we need it here, on the Icon comp it was because the button passes some default styles as array. 
not sure its the case here.

lidord-wix commented on src/components/image/index.web.tsx:
Please format the file :)

lidord-wix commented on src/components/image/index.web.tsx:
```diff
@@ -0,0 +1,236 @@
+    return null;
+  }
+
+  isGif() {
+    if (Constants.isAndroid) {
+      const {source} = this.props;
+      const url = _.get(source, 'uri');
+      const isGif = /(http(s?):)([/|.|\w|\s|-])*\.gif/.test(url ?? '');
+      return isGif;
+    }
+  }
```
I believe we can remove it here

lidord-wix commented on src/components/image/index.web.tsx:
Something looks different in the webDemo project in the `Animated Image` example, when I run it with the changes from this PR.

lidord-wix commented on src/components/image/index.web.tsx:
```diff
@@ -0,0 +1,236 @@
+
+    return (
+      // @ts-expect-error
+      <View style={imageViewStyle}>
```
Why do you add this extra `View` and not pass the style to the `img` component?

mendyEdri replied:
@OrenZak is right

lidord-wix reviewed (commented):
I added some comments, but... I have to say that I don't like this solution, it'll require Uilib to maintain two image files, and I'm confident that it will be forgotten. 
react-native-web supports RN's Image, so we should re-check what doesn't work and close the gaps. 
Until now we added the .web files only when something isn't supported on the web and I think we should keep doing it instead of duplicating to code.

mendyEdri commented:
> I added some comments, but... I have to say that I don't like this solution, it'll require Uilib to maintain two image files, and I'm confident that it will be forgotten. react-native-web supports RN's Image, so we should re-check what doesn't work and close the gaps. Until now we added the .web files only when something isn't supported on the web and I think we should keep doing it instead of duplicating to code.

I don't mind. atm in production sources with type of string (either source={url} or images actual data) are rendering the SVG component which causes issues and left us with production bugs.

Also, the current mobile image implementation cannot render ImageBackground properly in web.
moreover, using the mobile image, rendered styling weird, and wasn't affecting by image-ratio prop.

I suggest to assign someone from the team to address it :)
thanks!

OrenZak commented:
> > I added some comments, but... I have to say that I don't like this solution, it'll require Uilib to maintain two image files, and I'm confident that it will be forgotten. react-native-web supports RN's Image, so we should re-check what doesn't work and close the gaps. Until now we added the .web files only when something isn't supported on the web and I think we should keep doing it instead of duplicating to code.
> 
> I don't mind. atm in production sources with type of string (either source={url} or images actual data) are rendering the SVG component which causes issues and left us with production bugs.
> 
> Also, the current mobile image implementation cannot render ImageBackground properly in web. moreover, using the mobile image, rendered styling weird, and wasn't affecting by image-ratio prop.
> 
> I suggest to assign someone from the team to address it :) thanks!

@ethanshar can you help ?

ethanshar reviewed (commented):
I don't think we want to go with a duplicate file for the Image component just for web. 
These type of solutions won't scale, we can't maintain two identical files for two platforms.. 

Either we come up with a better solution, or please open a detailed ticket and we'll tailor this properly.

ethanshar commented:
Is this PR still relevant? can we close it?

mendyEdri commented:
> Is this PR still relevant? can we close it? 

It is. 
Removing it, still breaking images in web-editor

ethanshar commented:
But we don't use this fix, it's not merged. 
Is there still an issue with SVG?

mendyEdri commented:
> But we don't use this fix, it's not merged. 
> 
> Is there still an issue with SVG? 

We do, we use this version in web-editor.
And yes, there is still an issue

ethanshar commented:
@mendyEdri 
Is this PR still relevant?

mendyEdri commented:
> @mendyEdri 
> 
> Is this PR still relevant? 
> 
> 

Not anymore 💪
